inc/html.php

```diff
@@ -114,17 +114,9 @@ function html_secedit($text,$show=true){
  * @triggers HTML_SECEDIT_BUTTON
  */
 function html_secedit_button($matches){
-    $data = array('secid'  => $matches['secid'],
-        'target' => strtolower($matches['target']),
-        'range'  => $matches['range']);
-
-    if (!empty($matches['hid'])) {
-        $data['hid'] = strtolower($matches['hid']);
-    }
-
-    if (!empty($matches['name'])) {
-        $data['name'] = $matches['name'];
-    }
+    $data = json_decode($matches[1], true);
```

Comment:
We probably need to catch some error here to avoid problems where for some reason the old syntax is still used.

I don't think json_decode throws proper exceptions, so we probably need to suppress the error and check if $data is null.

Reply:
Alternatively, we could adjust the pattern matching the syntax to include the opening and closing `{}` of the JSON. That way we would ignore all old syntax.

Reply:
Is it likely that such errors can happen? I mean it is encoded in 'finishSectionEdit()', inc/parser/xhtml.php and decoded in 'html_secedit()', inc/html.php. I assume it could only happen if alternative renderers for HTML are used. Otherwise it is only DokuWiki core internal.

Reply:
Plugins might write their own section edit syntax to the HTML output without ever going through finishSectionEdit(). I don't know if any actually do though. I guess the easiest fix is to simply let the pattern look for the JSON delimiters.

Reply:
Ok, I did both: added the delimiters and checked for NULL anyway as a matching pattern alone cannot guarantee a proper JSON encoding.